### PR TITLE
TOC for .NET Design Guidelines

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
Added what amounts to a TOC for the full Azure SDK .NET specific guidelines. 

I would be especially interested in feedback what else we should cover.

### CODE REVIEW

File: docs/design/dotnet/DesignGuidelines.md [hunk 1]
```diff
@@ -1,16 +1,36 @@
...
+
+1.0.2 :warning: **NOTE** additional gudelines in this section overrule the Framework Design Guidelines  
+
+1.0.3 :white_check_mark: **DO** use Azure HTTP Pipeline to implement all methods that call Azure services. The pipeline takes care of many [General Azure SDK Guidelines](https://github.com/Azure/azure-sdk/blob/master/docs/design/common/DesignGuidelines.md), e.g. telemetry, logging, retries, mockability. Details of the pipeline design and usage are described in Appendix B below.
+
+# 2.0 Basic Azure SDK Design Pattern
+
+TODO:
+This scection describes the basic shape of a typical Azure SDK:
+- basic shape of the clinet types
```
Comment (johanste) [inline docs/design/dotnet/DesignGuidelines.md]: cli**ne**t -> client

File: docs/design/dotnet/DesignGuidelines.md [hunk 2]
```diff
@@ -19,3 +39,50 @@ i.e. send email about your proposed namespaces to this address to start a discus
...
+- spamples
+
+## 8.0 Common Type Usage
+TODO:
+- Etag
+- Response<T>
+- System.Uri
+  
+## Appendix A: Commonly Overlooked .NET API Design Guidelines
+Some .NET Design Guiudelines have been notoriously overlooked in exisitng Azure SDKs. This section serves as a way to highlight these guidelines.
```
Comment (johanste) [inline docs/design/dotnet/DesignGuidelines.md]: exis**it**ng -> existing

File: docs/design/dotnet/DesignGuidelines.md [hunk 3]
```diff
@@ -19,3 +39,50 @@ i.e. send email about your proposed namespaces to this address to start a discus
...
+
+:no_entry: **DO NOT** have abstractions (interfaces of abstract classes) unless types both implement and consume them (i.e. you have parameters typed as the abstraction).
```
Comment (JeffreyRichter) [inline docs/design/dotnet/DesignGuidelines.md]: Maybe give an example; I'm not sure what you mean by this.

File: docs/design/dotnet/DesignGuidelines.md [hunk 4]
```diff
@@ -19,3 +39,50 @@ i.e. send email about your proposed namespaces to this address to start a discus
...
+
+:no_entry: **DO NOT**  use interfaces if you can use abstract classes. The only reasons to use an interface are: a) you need to “multiple-inherit”, b) you want structs to implement and abstraction.
```
Comment (JeffreyRichter) [inline docs/design/dotnet/DesignGuidelines.md]: typo: you want structs to implement "an" abstraction

File: docs/design/dotnet/DesignGuidelines.md [hunk 5]
```diff
@@ -19,3 +39,50 @@ i.e. send email about your proposed namespaces to this address to start a discus
...
+
+ :no_entry: **DO NOT**  use very generic words and terms for type names, e.g. do not use names like “OperationResponse”, “DataCollection” etc.
+
+:warning: **AVOID** parameters types where it’s not clear what valid values are supported, e.g. string parameters are often like that.
```
Comment (JeffreyRichter) [inline docs/design/dotnet/DesignGuidelines.md]: typo: parameter types. Maybe give an example.

File: docs/design/dotnet/DesignGuidelines.md (hunk 1 above)
Comment (KrzysztofCwalina) [reply docs/design/dotnet/DesignGuidelines.md]: thanks!

File: docs/design/dotnet/DesignGuidelines.md (hunk 2 above)
Comment (KrzysztofCwalina) [reply docs/design/dotnet/DesignGuidelines.md]: thanks!

File: docs/design/dotnet/DesignGuidelines.md [hunk 6]
```diff
@@ -19,3 +39,50 @@ i.e. send email about your proposed namespaces to this address to start a discus
 - Azure.Runtime (e.g. Azure.Runtime.VirtualMachines)
 - Azure.Security (e.g. Azure.Security.KeyVault)
 - Azure.Storage (e.g. Azure.Storage.Blobs)
+
+## 3.1.0 Type Naming
+TODO: XxxClient, XxxResopurce, etc.
```
[the remark this answers is not in the document]
Comment (KrzysztofCwalina) [reply docs/design/dotnet/DesignGuidelines.md]: thanks!

File: docs/design/dotnet/DesignGuidelines.md (hunk 4 above)
Comment (KrzysztofCwalina) [reply docs/design/dotnet/DesignGuidelines.md]: see above. I define an "abstraction" as interface or abstract class.

File: docs/design/dotnet/DesignGuidelines.md (hunk 3 above)
Comment (KrzysztofCwalina) [reply docs/design/dotnet/DesignGuidelines.md]: the i.e. was supposed to be the example/explanation.

File: docs/design/dotnet/DesignGuidelines.md (hunk 5 above)
Comment (KrzysztofCwalina) [reply docs/design/dotnet/DesignGuidelines.md]: will do

**Pull request merged**